frame/evm/src/runner/stack.rs

```diff
@@ -187,13 +183,7 @@ where
 		R: Default,
 	{
 		// Used to record the external costs in the evm through the StackState implementation
-		let mut maybe_weight_info =
-			WeightInfo::new_from_weight_limit(weight_limit, proof_size_base_cost).map_err(
-				|_| RunnerError {
-					error: Error::<T>::GasLimitTooLow,
-					weight,
-				},
-			)?;
+		let mut maybe_weight_info = WeightInfo::new_from_weight_limit(weight_limit);
```

Comment:
Solochains will not have proof_size limits
```suggestion
                 let recording_proof_size = proof_size_pre_execution.is_some();
		let mut maybe_weight_info = WeightInfo::new_from_weight_limit(weight_limit, recording_proof_size);
```